crates/uv-distribution-types/src/file.rs

```diff
@@ -190,6 +191,18 @@ impl From<&Url> for UrlString {
     }
 }
 
+impl From<LogSafeUrl> for UrlString {
+    fn from(value: LogSafeUrl) -> Self {
+        Self(value.as_str().into())
+    }
+}
+
+impl From<&LogSafeUrl> for UrlString {
+    fn from(value: &LogSafeUrl) -> Self {
+        Self(value.as_str().into())
+    }
+}
+
 impl Display for UrlString {
```

Comment:
Do we need to apply redactions when displaying `UrlString`s?

Reply:
This is a good question. We probably should. I'm using `LogSafeUrl`s to create `UrlString`s in these changes wherever we were using `Url`'s, but you can also create one directly from a `&str`.